src/main.c

```diff
@@ -61,87 +58,58 @@
 #include "upnp_transport.h"
 #include "upnp_connmgr.h"
 
-static gboolean show_version = FALSE;
-static gboolean show_devicedesc = FALSE;
-static gboolean show_connmgr_scpd = FALSE;
-static gboolean show_control_scpd = FALSE;
-static gboolean show_transport_scpd = FALSE;
-static gboolean show_outputs = FALSE;
-static gboolean daemon_mode = FALSE;
+static int show_version = FALSE;
```

Comment:
Instead of changing gboolean to int -- which looses all the idea behind using them in the first place, some user-visible type that indicates what this is about -- I suggest to include `<stdbool.h>` and use the `bool` type, as well as the `true` and `false` constants that come with it.

There are quite a few places where you had to downgrade a gboolean to an int; with using stdbool.h, we can keep the sanity.